### ISSUE TITLE
conversion from `utf8' to `UCS-4LE' is not allowed

### ISSUE THREAD

**Issue by UziTech** [AtomLinter/linter-phpcs#232]
phpcs sends this error when trying to encode  utf8 because 'utf8' is not a valid encoding. it should be changed to 'utf-8'

I'm not sure if this is an issue that should be fixed by linter-phpcs or atom

**Comment by Arcanemagus** [comment]
If you change the code to force it to `utf-8` does it work? The error in the title seems to indicate a different problem.

**Comment by UziTech** [comment]
yes it does work with 'utf-8'

**Comment by UziTech** [comment]
my fix was to just change the encoding to 'utf-8' when it sees 'utf8'

**Comment by UziTech** [comment]
another fix could be to only add the `--encoding` argument if the encoding is not 'utf8'  sing phpcs uses that as a default

**Comment by UziTech** [comment]
nevermind it uses 'iso-8859-1' by default

**Comment by weirdan** [comment]
So it appears mapping for --encoding is actually needed. 

What's your phpcs version, @UziTech ? Mine phpcs works fine with --encoding=utf8, that's why I thought omitting the encoding name mapping might be ok.

**Comment by UziTech** [comment]
PHPCS 2.7.0 on windows 10 with PHP 5.6.20

**Comment by UziTech** [comment]
here is a list of the supported encodings https://www.gnu.org/software/libiconv/

**Comment by UziTech** [comment]
this seems to be the file where they define the encodings and aliases 

http://git.savannah.gnu.org/cgit/libiconv.git/tree/lib/encodings.def

**Comment by UziTech** [comment]
according to this bug report for Drupal in 2014 for the same issue, this only seems to be a windows issue

https://www.drupal.org/node/2368747

**Comment by weirdan** [comment]
Don't let the name confuse you, GNU libiconv is just one of several implementations. On linux iconv is provided by libc, for example.

**Comment by UziTech** [comment]
I see. From what I have gathered it seems like the best solution, although a bit more invasive, would be to get atom to conform to the [unofficial standard](https://stackoverflow.com/questions/809620/utf8-or-utf-8) of using 'utf-*'

I'm not sure exactly how we would start doing that but I have created an issue in the text-buffer repo about it. https://github.com/atom/text-buffer/issues/194

Doing some searches through the atom repos, it looks like a lot of tests on different packages are looking for 'utf8' so this might not be easy to change

**Comment by weirdan** [comment]
Here I compiled the map of encoding names understood by iconv-lite (used by atom) to their «canonic names» + the map of «canonic names» to their libiconv counterparts. Additionally I checked that iconv extension on my machine accepts it. @UziTech, can you check that the following script produces no errors for you?:
```php
$map = json_decode(file_get_contents('iconv-lite-to-libiconv.json.txt'));
foreach ($map as $enc) {
    if (!is_null($enc)) {
        iconv_strlen("", $enc);
    }
}
```

[iconv-lite.aliases.json.txt](https://github.com/AtomLinter/linter-phpcs/files/706691/iconv-lite.aliases.json.txt)
[iconv-lite-to-libiconv.json.txt](https://github.com/AtomLinter/linter-phpcs/files/706692/iconv-lite-to-libiconv.json.txt)

**Comment by UziTech** [comment]
doesn't through any errors 👍

**Comment by weirdan** [comment]
Sorry guys, this has been very turbulent week for me, but I finally got to fixing this. Please test.

**Comment by Arcanemagus** [comment]
@weirdan's fix for this has been published in v1.5.7 so I'm going to mark this as closed for now, hopefully for good 😛.
